### ISSUE TITLE
False negative on unused import of nested modules

### ISSUE THREAD

**Issue by draustin** [microsoft/pyright#9622]
Pyright does not report all unused imports of modules inside a package.

I have a namespace package `package` with three modules:
![Image](https://github.com/user-attachments/assets/c78ecd60-a701-4e39-9fdc-474812533759)

`unused-import-false-negative.py` only contains import statements.

```python
import package.module1 # Pylance reports that "package.module1" is not accessed
import package.module2 # No error reported.
import package.module3 # No error reported.
```
I would expect all three imports to be reported as unused.

Also, if I refer to `package.module2` or `package.module3`, then the unused import error `package.module1` goes away (i.e. Pyright/Pylance reports zero errors).

I got the same results after adding an `__init__.py` to `package`.

I get the same results using Pylance v2024.12.100 (pre-release) in VS Code and Pyright 1.1.391 on the command line.

**Comment by erictraut** [comment]
Thanks for the bug report. This took a while and was a difficult issue to fix given the way that multi-part import names are modeled in pyright. It will be addressed in the next release of pyright.

**Comment by draustin** [comment]
Thank you!
